### PR TITLE
Add all faces context parameter rules for transformation

### PR DESCRIPTION
**Pull request by volosied**
Pulled from https://myfaces.apache.org/#/core40 and added to existing list (but sorted).

for #21185

### CODE REVIEW

File: dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties [hunk 1]
```diff
@@ -31,20 +31,39 @@ javax.ws.rs.Application=jakarta.ws.rs.Application
...
+javax.faces.FACELETS_SKIP_COMMENTS=jakarta.faces.FACELETS_SKIP_COMMENTS
+javax.faces.FACELETS_SUFFIX=jakarta.faces.FACELETS_SUFFIX
+javax.faces.FACELETS_VIEW_MAPPINGS=jakarta.faces.FACELETS_VIEW_MAPPINGS
+javax.faces.FULL_STATE_SAVING_VIEW_IDS=jakarta.faces.FULL_STATE_SAVING_VIEW_IDS
+javax.faces.HONOR_CURRENT_COMPONENT_ATTRIBUTES=jakarta.faces.HONOR_CURRENT_COMPONENT_ATTRIBUTES
 javax.faces.INTERPRET_EMPTY_STRING_SUBMITTED_VALUES_AS_NULL=jakarta.faces.INTERPRET_EMPTY_STRING_SUBMITTED_VALUES_AS_NULL
-javax.faces.WEBAPP_CONTRACTS_DIRECTORY=jakarta.faces.WEBAPP_CONTRACTS_DIRECTORY
-javax.faces.CLIENT_WINDOW_MODE=jakarta.faces.CLIENT_WINDOW_MODE
+javax.faces.LIFECYCLE_ID=jakarta.faces.LIFECYCLE_ID
+javax.faces.NUMBER_OF_CLIENT_WINDOWS=jakarta.faces.NUMBER_OF_CLIENT_WINDOWS
```
Comment (volosied) [inline dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties]: Added in 4.0. Remove `javax.faces.NUMBER_OF_CLIENT_WINDOWS=jakarta.faces.NUMBER_OF_CLIENT_WINDOWS`

File: dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties [hunk 2]
```diff
@@ -31,20 +31,39 @@ javax.ws.rs.Application=jakarta.ws.rs.Application
...
 #Faces Configurations
-javax.faces.webapp.FacesServlet=jakarta.faces.webapp.FacesServlet
-javax.faces.PROJECT_STAGE=jakarta.faces.PROJECT_STAGE
-javax.faces.ENABLE_WEBSOCKET_ENDPOINT=jakarta.faces.ENABLE_WEBSOCKET_ENDPOINT
 javax.faces.ALWAYS_PERFORM_VALIDATION_WHEN_REQUIRED_IS_TRUE=jakarta.faces.ALWAYS_PERFORM_VALIDATION_WHEN_REQUIRED_IS_TRUE
-javax.faces.FACELETS_LIBRARIES=jakarta.faces.FACELETS_LIBRARIES
-javax.faces.WEBAPP_RESOURCES_DIRECTORY=jakarta.faces.WEBAPP_RESOURCES_DIRECTORY
-javax.faces.DISABLE_FACESSERVLET_TO_XHTML=jakarta.faces.DISABLE_FACESSERVLET_TO_XHTML
-javax.faces.VIEWROOT_PHASE_LISTENER_QUEUES_EXCEPTIONS=jakarta.faces.VIEWROOT_PHASE_LISTENER_QUEUES_EXCEPTIONS
+javax.faces.AUTOMATIC_EXTENSIONLESS_MAPPING=jakarta.faces.AUTOMATIC_EXTENSIONLESS_MAPPING
```
Comment (volosied) [inline dev/wlp-jakartaee-transform/rules/jakarta-xml-dd.properties]: `AUTOMATIC_EXTENSIONLESS_MAPPING` Added in 4.0 - remove rule

**Comment by pnicolucci** [comment]
Best way I know to validate this is looking: https://jakarta.ee/specifications/faces/2.3/apidocs/constant-values.html and searching for "PARAM_NAME" should give us a complete list to be concerned about!

I'm not aware of anything being removed in Faces 3.0 or added for that fact. So looking at 2.3 or 3.0 javadocs should be fine.

**Comment by volosied** [comment]
Only thing I'm not sure about is "javax.faces.SEPARATOR_CHAR" as this is listed as a spec parameter, but I don't see it configurable via MyFaces?

**Comment by volosied** [comment]
The jakarta.faces.SEPARATOR_CHAR param: It's configurable -- just tested on Faces 4.0 using the namespacedView TCK app.

```
    <context-param>
        <param-name>jakarta.faces.SEPARATOR_CHAR</param-name>
        <param-value>*</param-value>
    </context-param>
```
Generated element id: 
`j_id__v_0*panel1`

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
comment above was for the first build.

Current build is still in progress:  https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_dhhugLg6Ee2r6re_Zvbagg

**Comment by volosied** [comment]
#libby

**Pull request merged**